modules/nf-core/untar_files/main.nf

```diff
@@ -0,0 +1,63 @@
+process UNTAR_FILES {
+    tag "$archive"
+    label 'process_single'
+
+    conda "conda-forge::sed=4.7 bioconda::grep=3.4 conda-forge::tar=1.34"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/ubuntu:20.04' :
+        'ubuntu:20.04' }"
+
+    input:
+    tuple val(meta), path(archive)
+
+    output:
+    tuple val(meta), path("${prefix}/**") , emit: files, optional: true
+    path "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args  = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    prefix    = task.ext.prefix ?: ( meta.id ? "${meta.id}" : archive.baseName.toString().replaceFirst(/\.tar$/, ""))
+
+    """
+    mkdir $prefix
+
+    ## Ensures --strip-components only applied when top level of tar contents is a directory
+    ## If just files or multiple directories, place all in prefix
+    if [[ \$(tar -taf ${archive} | grep -o -P "^.*?\\/" | uniq | wc -l) -eq 1 ]]; then
+        tar \\
+            -C $prefix --strip-components 1 \\
```

Comment:
Shouldn't this be an `ext.args` argument? No need to make this module too complex (this should be something that should be resolved on a pipeline level?)